src/main/java/io/rocketpartners/cloud/utils/SqlUtils.java

```diff
@@ -431,6 +431,60 @@ public static String buildInsertSQL(Connection conn, String tableName, Object[]
       return sql.toString();
    }
 
+   public static String buildInsertOnDuplicateKeySQL(Connection conn, String tableName, Object[] columnNameArray)
+   {
+      StringBuffer sql = new StringBuffer(buildInsertSQL(conn, tableName, columnNameArray));
+      sql.append(" ON DUPLICATE KEY UPDATE ");
+      for (Object col : columnNameArray)
+      {
+         sql.append(col).append("=VALUES(").append(col).append("), ");
+      }
+      sql.substring(0, sql.length() - 1);
+      return sql.toString();
+   }
+
+   public static String mysqlUpsert(Connection conn, String tableName, List<Map<String, Object>> rows) throws Exception
+   {
+      LinkedHashSet keySet = new LinkedHashSet();
```

Comment:
This would benefit from the inclusion of a batch size parameter.

Reply:
So we need to **stmt.executeBatch()**, when the batch size reaches a specific BATCH_MAX, right?